### PR TITLE
Release 3.1.0

### PR DESCRIPTION
**Pull request by ehab-al-cko**
- [CHANGED] Rename `ChallengeResult` to `AuthenticationResult`
- [CHANGED] Deprecate 2 `authenticate` functions
- [ADDED] Add new `authenticate` function with callback value of Swift `Result` type which contain `AuthenticationResult` and `AuthenticationError` 
[Improvements] Treat on receiving `transStatus` in ARES/CRES (Frictionless/Challenged flow) as Authentication Completed and Authentication error as all runtime SDK errors and protocol errors for ARES/CRES
[Fix] Fix Logging SDK version

### CODE REVIEW

File: Checkout3DS.xcframework/ios-arm64/Checkout3DS.framework/Headers/Checkout3DS-Swift.h [hunk 1]
```diff
@@ -576,7 +576,7 @@ SWIFT_CLASS("_TtC11Checkout3DS18Checkout3DSService")
 ///
 /// \param error If authentication fails, an <code>NSError</code> indicating the reason for failure, otherwise <code>nil</code>.
 ///
-- (void)authenticateWithAuthenticationParameters:(Checkout3DSAuthenticationParameters * _Nonnull)authenticationParameters completionHandler:(void (^ _Nonnull)(NSError * _Nullable))completionHandler;
+- (void)authenticateWithAuthenticationParameters:(Checkout3DSAuthenticationParameters * _Nonnull)authenticationParameters completionHandler:(void (^ _Nonnull)(NSError * _Nullable))completionHandler SWIFT_DEPRECATED_MSG("Use authenticate function with compilation closure (Result<AuthenticationResult, AuthenticationError>) -> Void)");
```
Comment (okhan-okbay-cko) [inline Checkout3DS.xcframework/ios-arm64/Checkout3DS.framework/Headers/Checkout3DS-Swift.h:579]:
```suggestion
- (void)authenticateWithAuthenticationParameters:(Checkout3DSAuthenticationParameters * _Nonnull)authenticationParameters completionHandler:(void (^ _Nonnull)(NSError * _Nullable))completionHandler SWIFT_DEPRECATED_MSG("Use authenticate function with completion closure (Result<AuthenticationResult, AuthenticationError>) -> Void)");
```

File: README.md [hunk 2]
```diff
@@ -92,42 +110,43 @@ let authenticationParameters = AuthenticationParameters(
...
 ```swift
    private var transaction: Transaction?
    private var standalone3DSService: ThreeDS2Service?
 ````
 
-2. Initialize the SDK with your preferred user interface options:
+2-Initialize the SDK with your preferred user interface options:
 
-- `scheme` can be set only as `visa` and `mastercard` in lowercase string.
+- `scheme` can be set only as `visa` and `mastercard` in lower case string.
```
Comment (okhan-okbay-cko) [inline README.md:149]: on the other PR, @patrick-hoban-cko also suggested to use lowercase btw. I think that PR comment was resolved without being resolved

File: Checkout3DS.xcframework/ios-arm64/Checkout3DS.framework/Headers/Checkout3DS-Swift.h (hunk 1 above)
Comment (okhan-okbay-cko) [reply Checkout3DS.xcframework/ios-arm64/Checkout3DS.framework/Headers/Checkout3DS-Swift.h:579]: I didn't mark the others, thinking that they are all auto-generated. But this typo lives under other files as well. Just an FYI

**Comment by deepesh-vasthimal-cko** [comment]
Close the PR will create it via Bitrise workflow.

**Review by okhan-okbay-cko — APPROVED** [review]

**Pull request closed without merging**